lib/vault.py

```diff
@@ -185,51 +185,42 @@ def configure(self):
             # superset of any policies that it will provision
             provisioner_policies.append(name)
 
-        # Read AWS credentials file
-        vault_aws_creds = os.path.join(PRIVATE_DIR, "vault_aws_credentials")
-        if os.path.exists(vault_aws_creds):
-            with open(vault_aws_creds, "r") as fh:
-                aws_creds = json.load(fh)
-        else:
-            aws_creds = None
-
         # AWS Authentication Backend
-        if aws_creds is None:
-            print("Vault AWS credentials files does not exist, skipping configuration of AWS-EC2 authentication backend")
-        else:
+        #Enable AWS auth in Vault
+        if 'aws' not in client.list_auth_backends():
             try:
-                client.enable_auth_backend('aws-ec2')
-            except hvac.exceptions.InvalidRequest as ex:
-                print("aws-ec2 auth backend already created.")
-            client.write('auth/aws-ec2/config/client', access_key = aws_creds["aws_access_key"],
-                                                       secret_key = aws_creds["aws_secret_key"])
-
-            arn_prefix = 'arn:aws:iam::{}:instance-profile/'.format(aws_creds["aws_account"])
-            policies = [p for p in provisioner_policies if p not in ('provisioner',)]
-            for policy in policies:
-                client.write('/auth/aws-ec2/role/' + policy, policies = policy,
-                                                             bound_iam_role_arn = arn_prefix + policy)
-
-        # AWS Secret Backend
-        if aws_creds is None:
-            print("Vault AWS credentials file does not exist, skipping configuration of AWS secret backend")
+                client.enable_auth_backend('aws')
+            except Exception as e:
+                print("Error while enabling auth back end: " + e)
+                sys.exit(1)
```

Comment:
A Library should never call `sys.exit()` as it disrupts the caller. It should raise an exception or similar if it cannot complete the requested action.

Reply:
I'm going to tackle this today. And also test with the new Vault binary

Reply:
Done